contracts/MultiRewards.sol

```diff
@@ -535,28 +622,85 @@ contract MultiRewards is ReentrancyGuard, Pausable {
         rewardData[_rewardsToken].rewardsDuration = _rewardsDuration;
         emit RewardsDurationUpdated(_rewardsToken, rewardData[_rewardsToken].rewardsDuration);
     }
+    */
 
     /* ========== MODIFIERS ========== */
 
-    modifier updateReward(address account) {
-        for (uint i; i < rewardTokens.length; i++) {
-            address token = rewardTokens[i];
-            rewardData[token].rewardPerTokenStored = rewardPerToken(token);
-            rewardData[token].lastUpdateTime = lastTimeRewardApplicable(token);
-            if (account != address(0)) {
-                rewards[account][token] = earned(account, token);
-                userRewardPerTokenPaid[account][token] = rewardData[token].rewardPerTokenStored;
+    function _updateReward() internal {
+        IHypervisor(stakingToken).getReward();
```

Comment:
The Gamma staker contract appears to be calling a Gamma vault to trigger rewards to be transferred to the staker.

Reply:
this is fine. same would work with our vaults

Reply:
I double checked. Gamma does exactly what we plan to: rewards are collected during deposit, withdrawal, rebalance in the vault and during stake, unstake in the farming contract